### PR TITLE
documentation: the SageMaker distributed data parallel v1.4.0 release

### PR DESCRIPTION
**Pull request by mchoi8739**
*Description of changes:*
- Doc renewal for the the SageMaker distributed data parallel library v1.4.0 release.
- Added release note for SMDDP v1.4.0
- Made doc updates to the framework estimator classes to present correct distribution config options
- Archived the previous version of the doc
- Made structural changes

*Testing done:*
Locally built.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integ/test_workflow.py [hunk 1]
```diff
@@ -0,0 +1,3013 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (mchoi8739) [inline tests/integ/test_workflow.py:1]: Why does this integ test file keep showing up when I sync upstream dev?

**Review by Zha0q1 — APPROVED** [review]
Thank you!

**Review by roywei — APPROVED** [review]
LGTM, reviewed staging website as well. Thank you!

**Comment by mchoi8739** [comment]
@ahsan-z-khan why did you change the base branch to master? Isn't the master branch deprecated?

**Comment by ahsan-z-khan** [comment]
> @ahsan-z-khan why did you change the base branch to master? Isn't the master branch deprecated?

From today, we have started using master branch. as going forward all of the releases will be from master branch.

**Review by ahsan-z-khan — COMMENTED** [review]
Can you rebase so it shows only your commits?

**Comment by mchoi8739** [comment]
> Can you rebase so it shows only your commits?

I think your rebasing worked fine and I don't need to do it from my side again.

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**